scripts/test_permanent.ps1

```diff
@@ -0,0 +1,67 @@
+$cwd=(Get-Location).Path
+
+try {
+
+    cd([System.IO.Path]::GetDirectoryName((resolve-path "$PSScriptRoot")))
+
+    ./emsdk.ps1 install latest
+    ./emsdk.ps1 activate latest --permanently
+
+    $EMSDK_USER = [System.Environment]::GetEnvironmentVariable("EMSDK", "User")
+    $EM_CONFIG_USER = [System.Environment]::GetEnvironmentVariable("EM_CONFIG", "User")
+    $EMSDK_NODE_USER = [System.Environment]::GetEnvironmentVariable("EMSDK_NODE", "User")
+    $EMSDK_PYTHON_USER = [System.Environment]::GetEnvironmentVariable("EMSDK_PYTHON", "User")
+    $JAVA_HOME_USER = [System.Environment]::GetEnvironmentVariable("JAVA_HOME", "User")
+    $EM_CACHE_USER = [System.Environment]::GetEnvironmentVariable("EM_CACHE", "User")
+    $PATH_USER = [System.Environment]::GetEnvironmentVariable("PATH", "User")
+
+    if (!$EMSDK_USER) {
+        throw "EMSDK is not set for the user"
+    }
+    if (!$EM_CONFIG_USER) {
+        throw "EM_CONFIG_USER is not set for the user"
+    }
+    if (!$EMSDK_NODE_USER) {
+        throw "EMSDK_NODE is not set for the user"
+    }
+    if (!$JAVA_HOME_USER) {
+        throw "JAVA_HOME is not set for the user"
+    }
+    if (!$EMSDK_PYTHON_USER) {
+        throw "EMSDK_PYTHON is not set for the user"
+    }
+    if (!$EM_CACHE_USER) {
+        throw "EM_CACHE is not set for the user"
+    }
+
+
+    $path_split = $path.Split(';')
+
+    $EMSDK_Path_USER = $path_split | Where-Object { $_ -match 'emsdk' }
+    if (!$EMSDK_Path_USER) {
+        throw "No path is added!"
+    }
+    $EMSDK_NODE_Path_USER = $path_split | Where-Object { $_ -match 'emsdk\\node' }
+    if (!$EMSDK_NODE_Path_USER) {
+        throw "emsdk\node is not added to path."
+    }
+    $EMSDK_PYTHON_Path_USER = $path_split | Where-Object { $_ -match 'emsdk\\python' }
+    if (!$EMSDK_PYTHON_Path_USER) {
+        throw "emsdk\python is not added to path."
+    }
+    $EMSDK_JAVA_Path_USER = $path_split | Where-Object { $_ -match 'emsdk\\java' }
+    if (!$EMSDK_JAVA_Path_USER) {
+        throw "emsdk\java is not added to path."
+    }
+
+    $EMSDK_UPSTREAM_Path_USER = $path_split | Where-Object { $_ -match 'emsdk\\upstream\\emscripten' }
+    if (!$EMSDK_UPSTREAM_Path_USER) {
+        throw "emsdk\upstream\emscripten is not added to path."
+    }
+
+}
+finally
+{
+# return back to cwd
+cd $cwd
```

Comment:
If you don't do this then does the directory change effect the calling shell?

Reply:
This is so if the program encounters an error or the user press CTRL+C, the current working directory will be restored to where it was. This is a safe way to change the current working directory.

Reply:
Just to be clear, it sounds like the `cd` command effect *does* effect the calling process?   e.g. if I run `foo.sp1` can it change the current working directory in my shell or are the changes limited to the duration of the `foo.ps1` script?

Reply:
Yes. It does. It changes the current working directory of the parent process. That's why I need to use `finally`.

Reply:
Another way is to use `Push-Location` in `try` and then call `Pop-Location` in the `finally`. It is the same as my method.